framework/db/QueryBuilder.php

```diff
@@ -122,6 +122,15 @@ public function build($query, $params = [])
             $sql = "($sql){$this->separator}$union";
         }
 
+        // replace aliases with table names
+        $sql = preg_replace_callback(
+            '/\\{\\{@([\w\-\. ]+)\\}\\}/',
+            function ($matches) use ($query) {
+                return '{{' . $query->getAlias($matches[1]) . '}}';
+            },
+            $sql
+        );
```

Comment:
This looks weird. Why do we perform alias post-processing in SQL?
Can't we deal with aliases during SQL composition, instead of after it is complete?